src/coreclr/scripts/superpmi.py

```diff
@@ -1956,9 +1956,12 @@ def replay_with_throughput_diff(self):
 
                     logging.info("Total instructions executed by base: {}".format(base_instructions))
                     logging.info("Total instructions executed by diff: {}".format(diff_instructions))
-                    delta_instructions = diff_instructions - base_instructions
-                    logging.info("Total instructions executed delta: {} ({:.2%} of base)".format(delta_instructions, delta_instructions / base_instructions))
-                    tp_diffs.append((os.path.basename(mch_file), base_instructions, diff_instructions))
+                    if base_instructions != 0 and diff_instructions != 0:
+                        delta_instructions = diff_instructions - base_instructions
+                        logging.info("Total instructions executed delta: {} ({:.2%} of base)".format(delta_instructions, delta_instructions / base_instructions))
+                        tp_diffs.append((os.path.basename(mch_file), base_instructions, diff_instructions))
+                    else:
+                        logging.warning("One compilation failed to produce any results")
```

Comment:
Does asmdiffs need it too? When did we see this? Seems like things would need to be pretty catastrophically broken for it to happen.

Reply:
I don't think asmdiffs needs it (I didn't see any failures), but I could be wrong. Yes, the "catastrophically broken" case is that currently one of the OSX-arm64 collections has a single MC, but it fails to replay due to "MISSING" error.

Reply:
Can we just purge that single MC file? Or is it about to be overrwritten?

Reply:
> Can we just purge that single MC file? Or is it about to be overrwritten?

I prefer to never do hand fixes to the Azure Storage data. And you're right, it should have been overridden this weekend, but that particular osx-arm64 job failed.